lib/ts/recipe/anomalydetection/recipeImplementation.ts

```diff
@@ -0,0 +1,72 @@
+/* Copyright (c) 2021, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import NormalisedURLPath from "../../normalisedURLPath";
+import { Querier } from "../../querier";
+import { UserContext } from "../../types";
+import { RecipeInterface, TypeNormalisedInput } from "./types";
+import { getDeviceHeadersFromRequest, getIpAddressFromRequest } from "./utils";
+
+export default function getRecipeInterface(querier: Querier, config: TypeNormalisedInput): RecipeInterface {
+    return {
+        checkAnomaly: async function ({ ipAddress, headers, ...rest }) {
+            if (!ipAddress && !headers) {
+                // no headers or ip address, so we can't check for anomaly
+                return { status: "OK" };
+            }
+
+            let response = await querier.sendPostRequest(
+                new NormalisedURLPath("/anomaly/check"),
+                {
+                    ipAddress,
+                    headers,
+                    ...rest,
+                },
+                {} as UserContext
+            );
+
+            if (response.status === "OK") {
+                return {
+                    status: "OK",
+                };
+            } else {
+                return {
+                    status: "ANOMALY_DETECTED",
+                };
+            }
+        },
+        checkAnomalyWithRequest: async function ({ request, ...rest }) {
+            const headers = (this as RecipeInterface).getDeviceHeadersFromRequest({ request });
+            const ipAddress = (this as RecipeInterface).getIpAddressFromRequest({ request });
+            const result = await (this as RecipeInterface).checkAnomaly({
+                headers,
+                ipAddress,
+                ...rest,
+            });
+            if (config.mode === "REPORT_ONLY") {
+                return {
+                    status: "OK",
```

Comment:
we aren't really reporting here. Just ignoring it? Also, this doesn't actually take into account the result value.

Reply:
the anomaly service is doing the reporting, it always does, like an audit service. But it always return the real state. So here if the developer set as REPORT_ONLY we just passthrough the OK status.

Reply:
actually we don't need this if here since we are already checking it on the other recipies